lib/ddtrace/span_operation.rb

```diff
@@ -100,6 +106,12 @@ def initialize(
       start(start_time) if start_time
     end
 
+    # Resource string.
+    # This string will be converted to UTF-8 encoding if in a different encoding.
+    def resource=(resource)
```

Comment:
This seems fine. Is it worthwhile being proactive and applying it to other string attributes as well? Like `service`, `type`, `name`?

Reply:
There's a cost to doing the UTF-8 check.

Resource can be dynamically created (as it can represent some string extracted from a gem, like a sql query or endpoint path), but I don't see `service`, `type` and `name` as having dynamically created values that would be non-UTF-8.

Given invoking `Core::Utils.utf8_encode` has overhead, would it make sense to add it to 3 additional fields that I believe would never need this conversion?

Reply:
I'm not 100% attached to only encoding `resource`.
If we think it's best to play it safe here, despite the small additional overhead on span creation, I'm happy to make the change.

Reply:
If cost is a consideration, this is probably fine as is.